### DISCUSSION TITLE
How to use WebSocket before [Native WebSocket support #90] land

### DISCUSSION THREAD

**Discussion by axetroy** [rwf2/Rocket#2136] [Questions]
#90

I want to create a WebSocket server with https://crates.io/crates/tokio-tungstenite

But I can not find any document for this

**Comment by the10thWiz** [comment]
There are are a few solutions before Native Websocket Support lands. 

- Creating a websocket server on a separate port. This separates the servers, and allows them to coexist.
- Same as above, but placing Rocket and Tungstenite behind a reverse proxy like Nginx, so you can expose them on the same port.
- In some cases, Server Sent Events (SSE) may be a reasonable substitute for websockets. Rocket already supports this in `0.5.0-rc.1`.

I don't think there are any other reasonable solutions at this point.
